pkg/aws/helpers.go

```diff
@@ -457,8 +484,52 @@ func GetPrefixFromOperatorRole(cluster *cmv1.Cluster) string {
 	return rolePrefix
 }
 
-func GetAccountRoleName(cluster *cmv1.Cluster) (string, error) {
-	return GetResourceIdFromARN(cluster.AWS().STS().RoleARN())
+func GetOperatorRolePolicyPrefixFromCluster(cluster *cmv1.Cluster, awsClient Client) (string, error) {
+	operatorRoles := cluster.AWS().STS().OperatorIAMRoles()
+	for _, operatorRole := range operatorRoles {
+		roleName, _ := GetResourceIdFromARN(operatorRole.RoleARN())
+		policiesDetails, err := awsClient.GetAttachedPolicy(&roleName)
+		if err != nil {
+			return "", err
+		}
+		attachedPoliciesDetail := FindAllAttachedPolicyDetails(policiesDetails)
+		policyPrefixCountMap := make(map[string]int)
+		for _, attachedPolicyDetail := range attachedPoliciesDetail {
+			index := strings.LastIndex(attachedPolicyDetail.PolicyName, "-openshift")
+			if index != -1 {
+				if _, ok := policyPrefixCountMap[attachedPolicyDetail.PolicyName[0:index]]; !ok {
+					policyPrefixCountMap[attachedPolicyDetail.PolicyName[0:index]] = 0
+				}
+				policyPrefixCountMap[attachedPolicyDetail.PolicyName[0:index]]++
+			}
+		}
+		rankedPolicyPrefix := helper.RankMapStringInt(policyPrefixCountMap)
```

Comment:
This is worse performance wise, but safer to find 'correct' prefix when there are more than one that matches possible policy prefixes from attached policies

Reply:
NP
If we are interested only in the `correct` prefix, we could use here a **find max** function instead of sorting?
Though this is a code with no API calls, so I think the difference is not going to be significant.

Reply:
I don't think max works for maps, does it? Could use a max heap instead which would be more efficient. But there shouldn't be too many possible prefixes anyway, not sure if it is worth the work. What do you think?